### PR TITLE
Load Scalafix migrations from this repository

### PR DESCRIPTION
**Pull request by fthomas**
This changes `MigrationAlg` so that it loads always the latest Scalafix
migrations from this repository instead of using the static list of
migrations that is shipped as resource in the JAR.

The following changes with this PR:
* The `--scalafix-migrations` option can now be used multiple times to
  load extra migrations not only from one but multiple files.
* Arguments to the `--scalafix-migrations` cannot only be local files
  but also remote URLs.
* Disabling loading the default Scalafix migrations is now done via a
  new `--disable-default-scalafix-migrations` option instead of using
  `disableDefaults` in an extra migrations file.
* Scala Steward now prints how many Scalafix migrations it loaded at
  start-up.

Closes: #1603

### CODE REVIEW

File: docs/scalafix-migrations.md [hunk 1]
```diff
@@ -31,21 +31,21 @@ which resolves those rules and edits the source code accordingly.
...
 After you have written a new migration rule for a new version of your project,
 Scala Steward needs to be made aware of it. Creating a pull request that adds
 the new rule to the list of [migrations][migrations] is enough for that. Once
 that pull request is merged, Scala Steward will start using this migration.
-When running Scala Steward you can specify a file containing extra migrations
-that might not be present in the [default list][migrations].
-You can also specify if you want the default list to be disabled.
 
-The file is in [HOCON][HOCON] and should look like this:
+When running Scala Steward you can also specify files via the
```
Comment (mzuehlke) [inline docs/scalafix-migrations.md:44]:
```suggestion
When running Scala Steward you can also specify files or URLs via the
```

File: docs/scalafix-migrations.md [hunk 2]
```diff
@@ -31,21 +31,21 @@ which resolves those rules and edits the source code accordingly.
...
+`--scalafix-migrations` command-line option that contain additional
```
Comment (mzuehlke) [inline docs/scalafix-migrations.md:45]:
```suggestion
`--scalafix-migrations` command-line option that contains additional
```

File: docs/scalafix-migrations.md [hunk 3]
```diff
@@ -31,21 +31,21 @@ which resolves those rules and edits the source code accordingly.
...
+migrations which are not be present in the [default list][migrations].
```
Comment (mzuehlke) [inline docs/scalafix-migrations.md:46]:
```suggestion
migrations which are not present in the [default list][migrations].
```

**Review by mzuehlke — APPROVED** [review]

File: docs/scalafix-migrations.md (hunk 2 above)
Comment (fthomas) [reply docs/scalafix-migrations.md:45]: The subject this belongs to is "files or URLs".

**Pull request merged**